src/move/emojicoin_arena/sources/emojicoin_arena.move

```diff
@@ -77,32 +67,34 @@ module arena::emojicoin_arena {
         max_match_percentage: u64,
         /// Maximum amount of APT to match in octas, when locking in.
         max_match_amount: u64,
-        /// Amount of rewards that are available to claim for this melee while it is still active.
-        /// Measured in octas, conditional on vault balance.
+        /// Amount of rewards that are available to claim for this melee, measured in octas, and
+        /// conditional on vault balance. Reset to 0 when cranking the schedule.
         available_rewards: u64,
         /// All entrants who have entered the melee, used as a set.
         all_entrants: SmartTable<address, Nil>,
```

Comment:
Per @davidiw

> Also a quick hint, smart table has a potential denial of service is one of the buckets gets excessively used... as you know address generation can be particularly easy to manipulate within certain constraints, so would have to check the bucketing logic to determine how easily this could result in a broken or excessively expensive smart table.
Update: since the address is hard to fully generate and it goes through another hash (siphash), the impact is likely negligible.

Reply:
Resolved per removal of `SmartTable` structs